### PR TITLE
[test] add dummy test file

### PR DESCRIPTION
**Pull request by dglaeser**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ 
+jobs:
+  run-tests:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        py-version: ["3.9", "3.10", "3.11"]
+    
+    steps:
+      - uses: ./.github/actions/setup-dev-env
```
Comment (dglaeser) [inline .github/workflows/main.yml:20]:
```suggestion
      - uses: actions/checkout@v2
      - uses: ./.github/actions/setup-dev-env
```

File: .github/actions/setup-dev-env.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: setup-developer-environment
+runs:
+  using: "composite"
+  inputs:
+    python-version:
+      description: 'Specify the python version to use'
+      required: true
+  steps:
+    - uses: actions/checkout@v2
```
Comment (dglaeser) [inline .github/actions/setup-dev-env.yml:9]:
```suggestion
```

File: .github/actions/setup-dev-env/action.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+name: setup-developer-environment
+runs:
+  using: "composite"
+  inputs:
+    python-version:
+      description: 'Specify the python version to use'
+      required: true
```
Comment (dglaeser) [inline .github/actions/setup-dev-env/action.yml:7]:
```suggestion
inputs:
  python-version:
    description: 'Specify the python version to use'
    required: true
runs:
  using: "composite"
```

File: .github/actions/setup-dev-env/action.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    description: 'Specify the python version to use'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ inputs.python-version }}
+    - run: pip install -r requirements.txt
+      shell: bash
```
Comment (dglaeser) [inline .github/actions/setup-dev-env/action.yml:13]:
```suggestion
    - run: |
         pip install --upgrade pip
         pip install -r requirements.txt
      shell: bash
```

**Comment by dglaeser** [comment]
@SvenMarcus, @soerenPeters, what do you think about this as starting point for you in the tdd workshop?

I used `requirements.txt` over `pyproject.toml` just to keep it as simple as possible. Also, the CI does not test against an installed version of the package, but simply invokes `pytest` from the current directory. I thought this would be easiest to get going...

**Comment by soerenPeters** [comment]
> @SvenMarcus, @soerenPeters, what do you think about this as starting point for you in the tdd workshop?
> 
> I used `requirements.txt` over `pyproject.toml` just to keep it as simple as possible. Also, the CI does not test against an installed version of the package, but simply invokes `pytest` from the current directory. I thought this would be easiest to get going... 

Looks good!

> I saw that in your tdd workshop repo you first install the package, but then when invoking pytest from the top folder, which one is actually picked?

Not sure if I understand you correctly. We do not install the package but install the "dev" dependencies before invoking pytest. Similar to what you did.

**Comment by dglaeser** [comment]
> Not sure if I understand you correctly. We do not install the package but install the "dev" dependencies before invoking pytest. Similar to what you did.

Don't you do `pip install .[dev]`? I thought this installs the current package `.`, in the `[dev]` flavour? I quickly tried locally and if I type that in, it installs hvac into a venv.

**Comment by soerenPeters** [comment]
> > Not sure if I understand you correctly. We do not install the package but install the "dev" dependencies before invoking pytest. Similar to what you did.
> 
> Don't you do `pip install .[dev]`? I thought this installs the current package `.`, in the `[dev]` flavour? I quickly tried locally and if I type that in, it installs hvac into a venv. I mean in any case it'll end up doing the right thing I suppose, I was just wondering which of the two gets picked. But maybe I am misunderstanding here...

Ah ok, I believe you are right. I wasn't aware of that either. But I think it also fits the way you did it. Maybe @SvenMarcus  has a more professional opinion?

**Comment by SvenMarcus** [comment]
Yeah, you're correct. `pip install ".[dev]"` will install the package we're developing as well. That doesn't do any harm, though. 
The better command would actually be `pip install -e ".[dev]"`. This will just create symlinks to the package under development in the venv and therefore pick up any changes automatically.

**Comment by dglaeser** [comment]
> The better command would actually be pip install -e ".[dev]".
Yes, I think that's what pytest also recommends... just wasn't sure if that's going too deep and confuses people. But in order to show good practice that's what we maybe should be doing then..

**Pull request closed without merging**